### PR TITLE
Fixed address prefix + used ledger application for generic ledger

### PR DESCRIPTION
**Pull request by antonijzelinskij**
- Use Generic address prefix for Generic Ledger
- Chain address prefix for migration and legacy ledger
- Fixed used application for Generic address generation

### CODE REVIEW

File: feature-ledger-impl/src/main/java/io/novafoundation/nova/feature_ledger_impl/domain/account/common/selectAddress/SelectAddressLedgerInteractor.kt [hunk 1]
```diff
@@ -23,9 +23,9 @@ interface SelectAddressLedgerInteractor {
 
     suspend fun getDevice(deviceId: String): LedgerDevice
 
-    suspend fun loadLedgerAccount(chain: Chain, deviceId: String, accountIndex: Int): Result<LedgerAccountWithBalance>
+    suspend fun loadLedgerAccount(chain: Chain, deviceId: String, accountIndex: Int, ledgerVariant: LedgerVariant): Result<LedgerAccountWithBalance>
 
-    suspend fun verifyLedgerAccount(chain: Chain, deviceId: String, accountIndex: Int): Result<Unit>
+    suspend fun verifyLedgerAccount(chain: Chain, deviceId: String, accountIndex: Int, ledgerVariant: LedgerVariant): Result<Unit>
```
Comment (antonijzelinskij) [inline feature-ledger-impl/src/main/java/io/novafoundation/nova/feature_ledger_impl/domain/account/common/selectAddress/SelectAddressLedgerInteractor.kt:28]: Added ledgerVariant to support selecting address in Generic flow

File: feature-ledger-impl/src/main/java/io/novafoundation/nova/feature_ledger_impl/presentation/account/addChain/selectAddress/AddLedgerChainAccountSelectAddressViewModel.kt [hunk 2]
```diff
@@ -34,6 +35,8 @@ class AddLedgerChainAccountSelectAddressViewModel(
     messageFormatter = messageFormatter
 ) {
 
+    override val ledgerVariant: LedgerVariant = LedgerVariant.LEGACY
+
     override fun onAccountVerified(account: LedgerAccountWithBalance) {
```
Comment (antonijzelinskij) [inline feature-ledger-impl/src/main/java/io/novafoundation/nova/feature_ledger_impl/presentation/account/addChain/selectAddress/AddLedgerChainAccountSelectAddressViewModel.kt:40]: As I understand we can add Ledger chain accounts only for ledger legacy

File: feature-ledger-impl/src/main/java/io/novafoundation/nova/feature_ledger_impl/sdk/application/substrate/newApp/GenericSubstrateLedgerApplication.kt [hunk 3]
```diff
@@ -41,6 +43,10 @@ class GenericSubstrateLedgerApplication(
         return ledgerRepository.getGenericDerivationPath(metaId)
     }
 
+    override suspend fun getAddressPrefix(chainId: ChainId): Short {
+        return SS58Encoder.GENERIC_ADDRESS_PREFIX
+    }
+
```
Comment (antonijzelinskij) [inline feature-ledger-impl/src/main/java/io/novafoundation/nova/feature_ledger_impl/sdk/application/substrate/newApp/GenericSubstrateLedgerApplication.kt:49]: Always return generic address prefix (= 42) for Generic flow

File: feature-ledger-impl/src/main/java/io/novafoundation/nova/feature_ledger_impl/sdk/application/substrate/newApp/MigrationSubstrateLedgerApplication.kt [hunk 4]
```diff
@@ -25,6 +25,12 @@ class MigrationSubstrateLedgerApplication(
         return buildDerivationPath(applicationConfig.coin, accountIndex)
     }
 
+    override suspend fun getAddressPrefix(chainId: ChainId): Short {
+        val chain = chainRegistry.getChain(chainId)
+
+        return chain.addressPrefix.toShort()
+    }
+
```
Comment (antonijzelinskij) [inline feature-ledger-impl/src/main/java/io/novafoundation/nova/feature_ledger_impl/sdk/application/substrate/newApp/MigrationSubstrateLedgerApplication.kt:33]: Using chain address prefix for migration flow

**Review by valentunn — APPROVED** [review]

**Pull request merged**